Pilot/pilotCommands.py

```diff
@@ -861,13 +861,21 @@ def __setInProcessOpts(self):
       localUser = 'Unknown'
     self.log.info('User Name  = %s' % localUser)
     self.log.info('User Id    = %s' % localUid)
-    self.inProcessOpts = ['-s /Resources/Computing/CEDefaults']
-    self.inProcessOpts.append('-o WorkingDirectory=%s' % self.pp.workingDir)
-    self.inProcessOpts.append('-o /LocalSite/CPUTime=%s' % (int(self.pp.jobCPUReq)))
-    self.jobAgentOpts = ['-o MaxCycles=%s' % self.pp.maxCycles,
-                         '-o PollingTime=%s' % self.pp.pollingTime,
-                         '-o StopOnApplicationFailure=%s' % self.pp.stopOnApplicationFailure,
-                         '-o StopAfterFailedMatches=%s' % self.pp.stopAfterFailedMatches]
+    self.innerCEOpts = ['-s /Resources/Computing/CEDefaults']
+    self.innerCEOpts.append('-o WorkingDirectory=%s' % self.pp.workingDir)
+    self.innerCEOpts.append('-o /LocalSite/CPUTime=%s' % (int(self.pp.jobCPUReq)))
+    if self.pp.ceType == 'Pool':
+      self.jobAgentOpts = ['-o MaxCycles=%s' % max(self.pp.pilotProcessors, self.pp.maxCycles),
+                           '-o PollingTime=%s' % min(5, self.pp.maxCycles),
+                           '-o StopOnApplicationFailure=%s' % True and self.pp.stopOnApplicationFailure,
+                           '-o StopAfterFailedMatches=%s' % max(self.pp.pilotProcessors,
+                                                                self.pp.stopAfterFailedMatches),
+                           '-o FillingModeFlag=True']
```

Comment:
Wait, this was not present before? How did it work until now?

Reply:
It worked because, while the default in DIRAC is `False`, in LHCb configuration we override it (look for `FillingModeFlag` in the CS). This makes sure that this flag is always `True` for `PoolCE`

Reply:
Oh yes alright, great